### PR TITLE
Adding Channels API

### PR DESCRIPTION
**Pull request by grozensmith**
Added Channel Sample page. Updated docs to explain how channels work. 

Front logo [Front conversations](https://front.frontapp.com/open/top_124w3)

### CODE REVIEW

**Comment by grozensmith** [comment]
@travtechjones I've iterated through this a few times with a handful of engineers on the team. We'll keep iterating as devs use it, but this is a sufficient start! Please approve so we can merge :) 

cc @xdmnl

File: source/channel-sample.html.md [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+- Observe in Front that it is “sent” by the channel. Since we’re using the same `external_conversation_id`, this message threads with the message in Step 1.
+
+![Sending Messages 5](sending-messages-5.png)
+
+- Represents a message that was sent in your application that should be synced to Front.
+
+
+# General Notes
+- Notice the `webhook_url` is optional, but Front stringly recommends generating a unique webhook URL for each channel. 
+- Front threads messages bu the `external_conversation_id`
```
Comment (travtechjones) [inline source/channel-sample.html.md]: *by

File: source/channels-api.html.md [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+| `Message Receive` <br> `POST https://webhook.company.com/[unique_webhook]`     | This endpoint will be hit when a message is sent on your channel from Front. | [Message Payload](https://dev.frontapp.com/#messages) |
+| `Channel Delete` <br> `DELETE https://webhook.company.com/[unique_webhook]`<br>[optional to implement]   | This endpoint will be hit when a user deletes a channel of your channel type. All functionality is complete in Front without you implementing this, but it is optional in case you want to keep your channel's data in sync.         | None. Just a delete request. 
+
+##Security
+Front will sign the messages sent to your webhook server to guarantee that the request originated from Front. It is imperative you validate the message signature before accepting it into your system.
+
+The requests made to your webhook server will include 2 headers: `X-Front-Signature` and `X-Front-Request-Timestamp`. First, check that the timestamp says the request was made recently, to prevent replay attacks.. We recommend 5 minutes as the maximum time before rejecting a message.
```
Comment (travtechjones) [inline source/channels-api.html.md]: Just one `.` here.

File: source/channels-api.html.md [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+| `Channel Delete` <br> `DELETE https://webhook.company.com/[unique_webhook]`<br>[optional to implement]   | This endpoint will be hit when a user deletes a channel of your channel type. All functionality is complete in Front without you implementing this, but it is optional in case you want to keep your channel's data in sync.         | None. Just a delete request. 
+
+##Security
+Front will sign the messages sent to your webhook server to guarantee that the request originated from Front. It is imperative you validate the message signature before accepting it into your system.
+
+The requests made to your webhook server will include 2 headers: `X-Front-Signature` and `X-Front-Request-Timestamp`. First, check that the timestamp says the request was made recently, to prevent replay attacks.. We recommend 5 minutes as the maximum time before rejecting a message. 
+
+1. Retrieve the `X-Front-Request-Timestamp` header on the HTTP request, and the body of the request.
+2. Concatenate the timestamp, and the JSON stringified body of the request to form a string. Use a colon as the delimiter between the two elements.
```
Comment (travtechjones) [inline source/channels-api.html.md]: No comma after timestamp.

**Pull request merged**